Move some files to ATen/core

smessmer opened the pull request:
Summary: This diff takes ArrayRef.h and moves it to ATen/core, including all its dependencies.

Differential Revision: D9074144

ezyang commented:
It's really easy to avoid codemod on this PR, can we do so and leave the codemod, e.g., for the weekend? Instead of replacing `ATen/ArrayRef.h` with `ATen/core/ArrayRef.h`, keep the old header and just have it `#include <ATen/core/ArrayRef.h>`. As you have just seen, you're going to get lots of conflicts if you bundle the codemod along with moving the files along, and you will probably keep conflicting with every PR until you land.

smessmer committed:
Improve ArrayRef

Differential Revision: D8926785

fbshipit-source-id: 04f62dfe9ee9ff8c853efcceee0209fe552274aa

smessmer committed:
Remove implicit ArrayRef -> vector conversion

Differential Revision: D8961693

fbshipit-source-id: e2c8f37a30d8bbb8d3cd66053862b42e640e146b

smessmer committed:
Move C++17.h to ATen/core

Differential Revision: D9095209

fbshipit-source-id: 2f866ed5727513b013a0e27960c70a6d8bfc8e7e

smessmer committed:
Move some files to ATen/core (#10040)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/10040

This diff takes ArrayRef.h and moves it to ATen/core, including all its dependencies.

Differential Revision: D9074144

fbshipit-source-id: cd320f0924e2af2f4da235ad4fd7d3d372463c93
